scripts/upgrade-zulip

```diff
@@ -1,2 +1,4 @@
 #!/usr/bin/env bash
+set -e pipefail
```

Comment:
I'm removing this change.  While it may be an improvement, it needs its own commit explaining why it is correct, since the upgrade tool wrapper is sensitive code.

Reply:
We might want to borrow the other logic in `scripts/setup/install` for referencing `upgrade.log` as a useful resource?  Not sure.

Reply:
I think `upgrade-zulip-from-git` has the same bug.  

Can you fix these in a separate PR?  I'll want to backport it to the 4.x branch at least, and might consider doing a tiny 3.5 release for it as well in order to support this test.

Reply:
Yeah, Sure.